### ISSUE TITLE
DFA against targets that jumped will displace target to top of buildings

### ISSUE THREAD

**Issue by VampireSeraphin** [MegaMek/megamek#358]
[DFAtotopofbuilding.sav.gz](https://github.com/MegaMek/megamek/files/519999/DFAtotopofbuilding.sav.gz)

As funny as it was this time, I've seen this bug a couple times now. Victims of a DFA who are displaced into a building are moved to the top of the building instead of through it.

**Comment by Dylan-M** [comment]
That's because Mechs are :ninja:

**Comment by arlith** [comment]
Well, the saved game wasn't a whole lot of help, since it happened after the fact, and collapsed the building.  The only thing it really did was make me believe that there was actually a problem.  I wasn't able to reproduce the problem and I eventually had to start walking through the code blind.  Miraculously, I found the bug.

The issue occurs in `Entity.calcElevation`.  What happens is that, when the target is displaced into the building, `Server.doEntityDisplacement(4)` is called and around Server line 12074, the `Entity`'s current elevation is saved, and a new elevation is calculated with `Entity.calcElevation`, then if the `Entity` is displaced into a building, `Server.passBuildingWall` is called to process the PSR into the building.

`Entity.calcElevation` is used to determine what an `Entity` elevation will be when moving between a src hex and a dest hex, given a particular climb mode.  The climb mode in this case is always false.  However, when moving to a new hex with a building, climb mode is used to determine if the `Entity` should go on top or bottom, but also `Entity.isJumpingNow` is used.  If `Entity.isJumpingNow` is true, then the `Entity` goes on top of the building.

Turns out, `Entity.isJumpingNow` is only reset at the start of a new round.  So, if the `Entity` jumped in the move phase, `Entity.isJumpingNow` is still true, and during the DFA displacement.... up it goes.

The fix for this is simply to reset `Entity.isJumpingNow` in `Entity.newPhase`, since that state only matters during a single phase (movement).
